### PR TITLE
Add golangci-lint to workflows

### LINKED ISSUE #30: Add Linting in CI
See [golangci-lint](https://github.com/golangci/golangci-lint)

### PR DESCRIPTION
**Pull request by kavir1698**
Closes #30

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  # Exit code when at least one issue was found, default is 1
+  issues-exit-code: 1
+
+  # Include test files or not, default is true
+  tests: true
+
+linters:
+  # Enable specific linters
+  enable:
```
Comment (minottic) [inline .golangci.yml:13]: as discussed, just to note it down, going forward we might want to relax these lists and enable further checks. For now I would say it's fine. Thanks!

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+run:
+  # Timeout for analysis, e.g. 30s, 5m, default is 1m
+  deadline: 3m
```
Comment (minottic) [inline .golangci.yml:3]: do we need to increase the default?

**Review by minottic — COMMENTED** [review]
thansk! 2 very minor comments...

File: .golangci.yml (hunk 1 above)
Comment (kavir1698) [reply .golangci.yml:13]: Right. But some of the linters suggested things that didn't make sense. I tested each and every one and kept the ones whose errors could be fixed or whose error was important.

File: .golangci.yml (hunk 2 above)
Comment (kavir1698) [reply .golangci.yml:3]: 1m sounds too short. I already see errors after 28 seconds.

Comment (minottic) [reply .golangci.yml:3]: yes, makes sense. Just something to keep in mind and maybe review later

**Review by minottic — APPROVED** [review]

**Pull request merged**